### DISCUSSION TITLE
add public ca certification error

### DISCUSSION THREAD

**Discussion by Paper-Dragon** [ipxe/ipxe#1105] [General]
1.
i want add public ca to iso,my public ipxe deploy to cloudflare pages, must be use https.
i have alreay add ca certificate,but naver run successful.
i need help.
2.
If possible, I want to add all the public CAs, what should I do.
3.
and merry christmas to any one.
4.
build command
```bash
make -j4     EMBED=netboot.geekery.cn     DEBUG=httpcore,tls,x509     TRUST=/ansible/certs/ca-ipxe-org.crt,/ansible/certs/ca-netboot-geekery-cn.crt    
```

![图片](https://github.com/ipxe/ipxe/assets/49874457/be69e580-e50d-441e-934c-24efc0bf5ff9)

**Comment by Paper-Dragon** [comment]
Artifacts: https://github.com/Paper-Dragon/netboot.xyz/actions/runs/7284713668/artifacts/1128231736

**Comment by Paper-Dragon** [comment]
![图片](https://github.com/ipxe/ipxe/assets/49874457/67089976-bc06-4f0e-94b4-ff4e0bc2a878)

**Comment by Paper-Dragon** [comment]
My Domain root cert is X2
![图片](https://github.com/ipxe/ipxe/assets/49874457/7845f82c-69e0-4281-874a-38dd16bbb8c0)

How to add X2 root cert

i have add x2 pem from https://letsencrypt.org/zh-cn/certificates/

but donot running

**Comment by mcb30** [comment]
I think this may be an example of issue #606

**Reply by Paper-Dragon** [reply]
fixed. thank u, reslove the way is.
command
```bash
make -j8     EMBED=netboot.geekery.cn     DEBUG=httpcore,tls,x509     TRUST=/ansible/certs/ca-ipxe-org.crt,/ansible/certs/ca-netboot-geekery-cn.crt
```

file ca-netboot-geekery-cn.crt;
include
- https://letsencrypt.org/certs/isrgrootx1.pem
- https://letsencrypt.org/certs/lets-encrypt-r3.pem
copy two cert to file ca-netboot-geekery-cn.crt, r3 cert upper, rootx1 down.

example pipline and artifacts
https://github.com/Paper-Dragon/netboot.xyz/actions/runs/7288729644/job/19861862664

**Reply by Paper-Dragon** [reply]
oh,that is magic. this image can not be used today. i need help
